portea los ecigs de bay

Evankhell561 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
-portea los ecigs de bay, incluye minibaterias. y cartuchos de sabor.
<!-- Include a small to medium description of what your PR changes. Document every changes or this may delay review or even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
-because rol
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif of your feature if you want -->

## Changelog
:cl:
add: añade ecigs, cartuchos de ecigs y baterias especiales para ellos.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
ecigport

Commit:
ecigs v2

falta el "taste" de los sabores.
falta  acomodar los valores de la bateria.

Commit:
ecigs v3

fixeados los valores de la bateria, sus sprites, puedes saborear los ingredientes

Commit:
el barato solo alcanza para un  cartucho.

Commit:
ya no saboreas el agua si hay más ingredientes.

Commit:
oops

SagaStars commented:
![IMG_0402](https://user-images.githubusercontent.com/55407528/66793595-269be200-eecc-11e9-8191-684888867ea6.jpeg)

Evankhell561 commented:
¿Puedes creer que esta mierda me llevó todo un dia y el rework de teleciencias me llevó media hora?
¿Puedes creer que esta mierda me llevó como 15 testeos y los ia shell solo 5?

Commit:
consistencia material

Evankhell561 committed:
Merge branch 'master' into ecigs3

Commit:
maldito compilador online

Commit:
consistencia de baterias

-¡rating 7 wtf.!
-¿por qué iniciar con una bateria alta y luego crear una segunda derivada de la primera con menor energía para luego crear otra derivada de la primera con la misma carga que la primera? jódete, bay.

Commit:
fix y emp

-hay más casos donde se actualiza el icono del ecig.
-si un emp le impata, drena energía de la bateria.

iRyzor commented on code/HISPANIA/game/objects/items/weapons/ecigs.dm:
```diff
@@ -0,0 +1,314 @@
+			for (var/datum/reagent/R in ec_cartridge.reagents.reagent_list)
+				ec_cartridge.reagents.trans_id_to(C, R.id, max(REAGENTS_METABOLISM / (2 * ec_cartridge.reagents.reagent_list.len), 0.05)) //transfer at least .1 of each chem
+				if(idletaste >= idletaste_treshold)
+					if(ec_cartridge.reagents.reagent_list.len == 1) //si solo hay un ingrediente puedes saborear agua
+						to_chat(C, "<span class='notice'>You can taste [R.taste_description].</span>")
+					else if(R.id != "water") //no saboreas el agua si hay m�s ingredientes
+						to_chat(C, "<span class='notice'>You can taste [R.taste_description].</span>")
```
No entiendo el sentido de esto.

Evankhell561 replied:
Si no pongo eso estarás saboreando "agua" todo el rato. No creo que "agua" sea algo que deba ser saboreado cuando está mezclada con cafe.

iRyzor commented on code/HISPANIA/modules/power/cell.dm:
```diff
@@ -46,3 +46,28 @@
+/obj/item/stock_parts/cell/device
+	name = "standard device power cell"
+	desc = "A small power cell designed to power handheld devices."
+	icon = 'icons/HISPANIA/obj/power.dmi'
```
minus

iRyzor commented on code/HISPANIA/game/objects/items/weapons/ecigs.dm:
```diff
@@ -0,0 +1,314 @@
+/obj/item/clothing/mask/cigarette/ecig
+	name = "electronic cigarette"
+	desc = "Device with modern approach to smoking."
+	icon = 'icons/hispania/obj/clothing/masks.dmi'
```
Esto pasa a ser innecesario cuando se usa hispania_icons.

iRyzor commented on code/HISPANIA/game/objects/items/weapons/ecigs.dm:
```diff
@@ -0,0 +1,314 @@
+	var/idletaste_treshold = 10
+	hispania_icon = TRUE
+	sprite_sheets = list(
+		"Human" = 'icons/hispania/mob//mask.dmi',
+		"Unathi" = 'icons/hispania/mob//mask.dmi',
+		"Tajaran" = 'icons/hispania/mob//mask.dmi',
+		"Skrell" = 'icons/hispania/mob//mask.dmi',
+		"Vox" = 'icons/hispania/mob//mask.dmi',
+		"Diona" = 'icons/hispania/mob//mask.dmi',
+		"Drask" = 'icons/hispania/mob//mask.dmi',
+		"Grey" = 'icons/hispania/mob//mask.dmi',
+		"Machine"= 'icons/hispania/mob//mask.dmi'
```
Por qué doble // ?

iRyzor commented on code/HISPANIA/game/objects/items/weapons/ecigs.dm:
```diff
@@ -0,0 +1,314 @@
+	else
+		to_chat(user,"<span class='notice'>There's no cartridge connected.</span>")
+
+/obj/item/clothing/mask/cigarette/ecig/proc/Deactivate()
```
minus

iRyzor reviewed (commented):
- Sólo hay sprite en la boca cuando el cigarro está encendido (Esto solo pasa con el simple)
- Mil errores en consola, miratelo.
- No genera luz cuando está encendido
- He fumado gasolina sin ninguna consecuencia realmente peligrosa, ¿No debería explotar?. Lo mismo pasa con el resto de químicos inflamables.
